### In this notebook we will prepare the labels to be used in the neural net 

#### Three label sets will be made:

#### 1. Label set for active/inactive - classification

#### 2. Label set for pubchem activity score - regression

#### 3. Label set for inhibition primary at 6uM (IP6M) 

#### Import required libraries

In [1]:
import pandas as pd
import numpy as np

#### Pre-processing:

In [2]:
df = pd.read_csv('AID_604_datatable.csv', header= 0)
df.head(10)

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
0        RESULT_TYPE          NaN          NaN                      NaN   
1       RESULT_DESCR          NaN          NaN                      NaN   
2        RESULT_UNIT          NaN          NaN                      NaN   
3                  1     842122.0    6602571.0                 Inactive   
4                  2     842123.0    6602616.0                 Inactive   
5                  3     842124.0     644371.0                 Inactive   
6                  4     842125.0    6603132.0                 Inactive   
7                  5     842126.0    2850911.0                 Inactive   
8                  6     842127.0    6603374.0                 Inactive   
9                  7     842128.0    6603127.0                 Inactive   

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
0                     NaN                   NaN                        NaN   
1                     NaN                   NaN                        NaN   
2                     NaN                   NaN                        NaN   
3                     0.0                   NaN                        NaN   
4                     0.0                   NaN                        NaN   
5                     3.0                   NaN                        NaN   
6                     0.0                   NaN                        NaN   
7                     1.0                   NaN                        NaN   
8                     2.0                   NaN                        NaN   
9                     0.0                   NaN                        NaN   

                          Inhibition_Primary at 6 uM  
0                                              FLOAT  
1  %Inhibition in the primary screen at 6 micromo...  
2                                               NONE  
3                                                  0  
4                                               -0.3  
5                                                6.7  
6                                                1.6  
7                                                2.5  
8                                                3.6  
9                                                0.2

we can see that the first 3 rows of the table do not provide us with actual data so we will drop them 

In [3]:
df = df.drop([0,1,2])
df.head(10)

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
3                   1     842122.0    6602571.0                 Inactive   
4                   2     842123.0    6602616.0                 Inactive   
5                   3     842124.0     644371.0                 Inactive   
6                   4     842125.0    6603132.0                 Inactive   
7                   5     842126.0    2850911.0                 Inactive   
8                   6     842127.0    6603374.0                 Inactive   
9                   7     842128.0    6603127.0                 Inactive   
10                  8     842129.0     644380.0                 Inactive   
11                  9     842131.0    6602565.0                 Inactive   
12                 10     842132.0    6603080.0                 Inactive   

    PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
3                      0.0                   NaN                        NaN   
4                      0.0                   NaN                        NaN   
5                      3.0                   NaN                        NaN   
6                      0.0                   NaN                        NaN   
7                      1.0                   NaN                        NaN   
8                      2.0                   NaN                        NaN   
9                      0.0                   NaN                        NaN   
10                     0.0                   NaN                        NaN   
11                     1.0                   NaN                        NaN   
12                     0.0                   NaN                        NaN   

   Inhibition_Primary at 6 uM  
3                           0  
4                        -0.3  
5                         6.7  
6                         1.6  
7                         2.5  
8                         3.6  
9                         0.2  
10                        0.7  
11                          3  
12                        0.9

as our images are in order of increasing CID value, we must reorder our dataset in a corresponding manner 

In [4]:
df = df.sort_values(['PUBCHEM_CID'], ascending=True)

In [5]:
df.head()

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
29178              29176    4243530.0        525.0                 Inactive   
10810              10808     855876.0       1329.0                 Inactive   
10719              10717     855784.0       1676.0                 Inactive   
10551              10549     855615.0       1688.0                 Inactive   
10563              10561     855627.0       1780.0                 Inactive   

       PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  \
29178                     0.0                   NaN   
10810                     0.0                   NaN   
10719                     0.0                   NaN   
10551                     4.0                   NaN   
10563                     0.0                   NaN   

       PUBCHEM_ASSAYDATA_COMMENT Inhibition_Primary at 6 uM  
29178                        NaN                       -3.2  
10810                        NaN                        0.9  
10719                        NaN                       -0.9  
10551                        NaN                        7.2  
10563                        NaN                       -2.3

now we have to reindex 

In [6]:
df = df.reset_index(drop=True)
df.head(10)

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
0              29176    4243530.0        525.0                 Inactive   
1              10808     855876.0       1329.0                 Inactive   
2              10717     855784.0       1676.0                 Inactive   
3              10549     855615.0       1688.0                 Inactive   
4              10561     855627.0       1780.0                 Inactive   
5              14170     859279.0       1820.0                 Inactive   
6              10625     855691.0       1820.0                 Inactive   
7              10805     855873.0       1832.0                 Inactive   
8              10651     855717.0       1892.0                 Inactive   
9              10529     855595.0       1893.0                 Inactive   

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
0                     0.0                   NaN                        NaN   
1                     0.0                   NaN                        NaN   
2                     0.0                   NaN                        NaN   
3                     4.0                   NaN                        NaN   
4                     0.0                   NaN                        NaN   
5                     0.0                   NaN                        NaN   
6                     0.0                   NaN                        NaN   
7                     0.0                   NaN                        NaN   
8                     0.0                   NaN                        NaN   
9                     0.0                   NaN                        NaN   

  Inhibition_Primary at 6 uM  
0                       -3.2  
1                        0.9  
2                       -0.9  
3                        7.2  
4                       -2.3  
5                       -1.2  
6                        1.6  
7                        0.6  
8                       -1.7  
9                      -12.2

In [7]:
df.describe()

PUBCHEM_SID   PUBCHEM_CID  PUBCHEM_ACTIVITY_SCORE  \
count  5.980500e+04  5.980400e+04            59805.000000   
mean   3.770893e+06  2.398077e+06                1.561609   
std    2.569843e+06  3.734900e+06                2.355936   
min    8.421220e+05  5.250000e+02                0.000000   
25%    8.600630e+05  6.622392e+05                0.000000   
50%    4.244429e+06  1.590369e+06                1.000000   
75%    4.263011e+06  3.238147e+06                3.000000   
max    7.978459e+06  5.664295e+07              100.000000   

       PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  
count                   0.0                        0.0  
mean                    NaN                        NaN  
std                     NaN                        NaN  
min                     NaN                        NaN  
25%                     NaN                        NaN  
50%                     NaN                        NaN  
75%                     NaN                        NaN  
max                     NaN                        NaN

In [8]:
df.shape

(59805, 8)

#### Since the NN will only run on the first 10,000 datapoints, we will remove everything after the first 10,000 entries in the datatable 

In [9]:
df1 = df[:10000]
df1.shape

(10000, 8)

## Label Set 1: 
note* need to rearrange table in order to increasign PUBCHEM_CID

In [10]:
lbl1 = df1
print(str(lbl1.shape))
lbl1.head()

(10000, 8)


PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
0              29176    4243530.0        525.0                 Inactive   
1              10808     855876.0       1329.0                 Inactive   
2              10717     855784.0       1676.0                 Inactive   
3              10549     855615.0       1688.0                 Inactive   
4              10561     855627.0       1780.0                 Inactive   

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
0                     0.0                   NaN                        NaN   
1                     0.0                   NaN                        NaN   
2                     0.0                   NaN                        NaN   
3                     4.0                   NaN                        NaN   
4                     0.0                   NaN                        NaN   

  Inhibition_Primary at 6 uM  
0                       -3.2  
1                        0.9  
2                       -0.9  
3                        7.2  
4                       -2.3

Let's convert the active/inactive labels to binary 0 and 1s, where 1 = active and 0 = inactive 

In [11]:
lbl1 = lbl1.replace(['Inactive','Active'],[0,1])
lbl1.head()

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID  PUBCHEM_ACTIVITY_OUTCOME  \
0              29176    4243530.0        525.0                         0   
1              10808     855876.0       1329.0                         0   
2              10717     855784.0       1676.0                         0   
3              10549     855615.0       1688.0                         0   
4              10561     855627.0       1780.0                         0   

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
0                     0.0                   NaN                        NaN   
1                     0.0                   NaN                        NaN   
2                     0.0                   NaN                        NaN   
3                     4.0                   NaN                        NaN   
4                     0.0                   NaN                        NaN   

  Inhibition_Primary at 6 uM  
0                       -3.2  
1                        0.9  
2                       -0.9  
3                        7.2  
4                       -2.3

In [12]:
lbl1.describe()

PUBCHEM_SID    PUBCHEM_CID  PUBCHEM_ACTIVITY_OUTCOME  \
count  1.000000e+04   10000.000000               10000.00000   
mean   1.382626e+06  581454.673200                   0.00540   
std    1.639294e+06  171367.003898                   0.07329   
min    8.421240e+05     525.000000                   0.00000   
25%    8.451678e+05  644905.750000                   0.00000   
50%    8.482285e+05  648151.500000                   0.00000   
75%    8.512872e+05  651410.250000                   0.00000   
max    7.978450e+06  654711.000000                   1.00000   

       PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  
count            10000.000000                   0.0                        0.0  
mean                 1.436100                   NaN                        NaN  
std                  2.688911                   NaN                        NaN  
min                  0.000000                   NaN                        NaN  
25%                  0.000000                   NaN                        NaN  
50%                  1.000000                   NaN                        NaN  
75%                  2.000000                   NaN                        NaN  
max                 86.000000                   NaN                        NaN

Since the mean value of the activity column is 0.00540 and the total number of entries is 10,000, we can use math to determine there are a total of .0054x10000 = 54 active compounds in our dataset

As this number is relatively small, we must append more/the rest of the active compounds from the extended dataset to this subset

We can also apply transformations to our active compounds images to artificially increase the amount as well

#### 1.2 Assemble a small dataset with the active compounds from the 10,001 row and on

In [13]:
df2 = df[10000:]
df2.shape

(49805, 8)

In [14]:
df1.tail()

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
9995               8972     852103.0     654707.0                 Inactive   
9996               8973     852104.0     654708.0                 Inactive   
9997               8974     852105.0     654709.0                 Inactive   
9998               8975     852106.0     654710.0                 Inactive   
9999               8976     852107.0     654711.0                 Inactive   

      PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
9995                     1.0                   NaN                        NaN   
9996                     0.0                   NaN                        NaN   
9997                     0.0                   NaN                        NaN   
9998                     2.0                   NaN                        NaN   
9999                     0.0                   NaN                        NaN   

     Inhibition_Primary at 6 uM  
9995                        1.9  
9996                        0.3  
9997                          0  
9998                        3.8  
9999                        1.1

In [15]:
df2.head(10)

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
10000               8977     852108.0     654712.0                 Inactive   
10001               8978     852109.0     654713.0                 Inactive   
10002               8979     852110.0     654714.0                 Inactive   
10003               8980     852112.0     654716.0                 Inactive   
10004               8982     852114.0     654718.0                 Inactive   
10005               8983     852115.0     654719.0                 Inactive   
10006               8984     852116.0     654720.0                 Inactive   
10007               8985     852117.0     654721.0                 Inactive   
10008               8986     852118.0     654722.0                 Inactive   
10009               8992     852124.0     654733.0                 Inactive   

       PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  \
10000                     0.0                   NaN   
10001                     2.0                   NaN   
10002                     6.0                   NaN   
10003                     0.0                   NaN   
10004                     0.0                   NaN   
10005                     0.0                   NaN   
10006                     2.0                   NaN   
10007                     0.0                   NaN   
10008                     0.0                   NaN   
10009                     3.0                   NaN   

       PUBCHEM_ASSAYDATA_COMMENT Inhibition_Primary at 6 uM  
10000                        NaN                        1.3  
10001                        NaN                        3.5  
10002                        NaN                       10.4  
10003                        NaN                       -7.6  
10004                        NaN                       -0.9  
10005                        NaN                       -0.9  
10006                        NaN                        3.7  
10007                        NaN                          1  
10008                        NaN                        1.5  
10009                        NaN                        6.7

we can see a clean continuation between the two tables which means df2 does in fact contain the 10,001th and on row 

In [16]:
table_end = len(df2["PUBCHEM_ACTIVITY_OUTCOME"])
table_end = table_end + 10000  #since the index starts at 10000, we will add 10000 to the length so the entire table is
                               #read through


active_set = {}

for i in range(10000, table_end):
    if df2["PUBCHEM_ACTIVITY_OUTCOME"][i] == "Active":
        CID = df2["PUBCHEM_CID"][i]
        outcome = df2["PUBCHEM_ACTIVITY_OUTCOME"][i]
        active_set[CID] = outcome
print(str(len(active_set)))    

158


The total amount of active compounds in this dataset was 212 and 158 + 54 = 212 so all active compounds are accounted for. Now we will create a pandas dataframe from the "active_set" python dict and then append that dataframe to our lbl1 dataframe 

In [43]:
comp_ids_list =  []
activity_list = []

for key, value in active_set.items():
    comp_ids_list.append(key)
    activity_list.append(value)
#########
#for future use, I will extract the list of active compounds
import pickle

pickle.dump(comp_ids_list, open("list_of_active_cmps_158.p", "wb"))

########
active_set_data = {'PUBCHEM_CID': comp_ids_list,'PUBCHEM_ACTIVITY_OUTCOME': activity_list}
active_df_activity = pd.DataFrame(active_set_data)
#active_df_acitivty = active_df_activity[['PUBCHEM_CID','PUBCHEM_ACTIVITY_OUTCOME']] #switch order of columns to match lbls df
print(active_df_activity.head())
print(active_df_activity.shape)

  PUBCHEM_ACTIVITY_OUTCOME  PUBCHEM_CID
0                   Active    2533120.0
1                   Active    3238146.0
2                   Active    3237123.0
3                   Active    5739478.0
4                   Active    3236615.0
(158, 2)


and let's have the format of the active/inactive label match between dataframes

In [18]:
active_df_activity = active_df_activity.replace(['Active'],[1])
active_df_activity.head()

PUBCHEM_ACTIVITY_OUTCOME  PUBCHEM_CID
0                         1    2533120.0
1                         1    3238146.0
2                         1    3237123.0
3                         1    5739478.0
4                         1    3236615.0

#### Now to combine dataframes lbl2 and active_df_activity in a dataframe called "lbl1_final"

But first we need to drop the unnecessary columns from the lbls dataframe so that the dimensions of the two dataframes match 

In [19]:
subset_lbl1 = ["PUBCHEM_ACTIVITY_OUTCOME","PUBCHEM_CID"] #subset of columns from lbl1 to keep

lbl1_2 = lbl1[subset_lbl1] #copy the kept columns to a new df called "lbl1_2" - Lbl2 is for Label set 2
print(lbl1_2.head()) #examine the df to confirm desired transformation took place"
print(lbl1_2.shape)

   PUBCHEM_ACTIVITY_OUTCOME  PUBCHEM_CID
0                         0        525.0
1                         0       1329.0
2                         0       1676.0
3                         0       1688.0
4                         0       1780.0
(10000, 2)


now to combine them:

In [20]:
lbl1_final = lbl1_2.append(active_df_activity)
print(lbl1_final.shape)
print(lbl1_final.head())

(10158, 2)
   PUBCHEM_ACTIVITY_OUTCOME  PUBCHEM_CID
0                         0        525.0
1                         0       1329.0
2                         0       1676.0
3                         0       1688.0
4                         0       1780.0


#### Now we can export the list of activity outcomes from lbl1_final to be our labels for label set 1 (labels for the classification task) 

In [21]:
lbl1_final_export = lbl1_final['PUBCHEM_ACTIVITY_OUTCOME']

lbl1_final_export.to_csv('labels_set_1.csv')

need to convert the csv to list once imported to the final cNN notebook

## Label Set 2:

We will be starting with a copy of "df1" which was the a dataframe consisting of the first 10,000 rows from the complete dataset 

In [22]:
lbl2 = df1
lbl2.head()

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
0              29176    4243530.0        525.0                 Inactive   
1              10808     855876.0       1329.0                 Inactive   
2              10717     855784.0       1676.0                 Inactive   
3              10549     855615.0       1688.0                 Inactive   
4              10561     855627.0       1780.0                 Inactive   

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
0                     0.0                   NaN                        NaN   
1                     0.0                   NaN                        NaN   
2                     0.0                   NaN                        NaN   
3                     4.0                   NaN                        NaN   
4                     0.0                   NaN                        NaN   

  Inhibition_Primary at 6 uM  
0                       -3.2  
1                        0.9  
2                       -0.9  
3                        7.2  
4                       -2.3

In [23]:
lbl2.shape

(10000, 8)

#### Now let's take the subset of the dataframe that we're interested in, namely the PUBCHEM_ACTIVITY_SCORE column and the PUBCHEM_CID column

In [24]:
subset_lbl2 = ["PUBCHEM_ACTIVITY_SCORE","PUBCHEM_CID"]
lbl2 = lbl2[subset_lbl2] #copy the kept columns to a new df called "lbl1_2" - Lbl2 is for Label set 2
print(lbl2.head()) #examine the df to confirm desired transformation took place"
print(lbl2.shape)

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_CID
0                     0.0        525.0
1                     0.0       1329.0
2                     0.0       1676.0
3                     4.0       1688.0
4                     0.0       1780.0
(10000, 2)


#### Now to assemble the corresponding "active" dataframe to append to this "PUBCHEM_ACTIVITY_SCORE" dataframe 

In [25]:
#Reusing code from earlier on the notebook 

#assemble the dictionary of active compound CID: Pubchem_activity_score
pscore_active_dict = {}  #pscore = pubchem_active_score 

#This is just like what was done earlier for the creation of the dictionary (which ultimately was converted to a dataframe)
#only this time we are using the Pubchem_activity_score column rather than the pubchem_acivity_outcome column

#table_end is defined above (length of column + 10000)


for i in range(10000, table_end):
    if df2["PUBCHEM_ACTIVITY_OUTCOME"][i] == "Active":
        CID = df2["PUBCHEM_CID"][i]
        pub_score = df2["PUBCHEM_ACTIVITY_SCORE"][i]
        pscore_active_dict[CID] = pub_score
        
print(str(len(pscore_active_dict)))  

158


In [26]:
comp_ids_list =  []
pscore_list = []

for key, value in pscore_active_dict.items():
    comp_ids_list.append(key)
    pscore_list.append(value)

pscore_set_data = {'PUBCHEM_CID': comp_ids_list,'PUBCHEM_ACTIVITY_SCORE': pscore_list}
active_df_pscore = pd.DataFrame(pscore_set_data) #df of active compounds, df contains CIDs and pscores


print(active_df_pscore.head())
print(active_df_pscore.shape)

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_CID
0                    19.0    2533120.0
1                    76.0    3238146.0
2                    34.0    3237123.0
3                    16.0    5739478.0
4                    50.0    3236615.0
(158, 2)


#### Now to combine the two datasets (lbl2 + active_df_pscore) 

In [27]:
lbl2_final = lbl2.append(active_df_pscore)
print(lbl2_final.shape)
print(lbl2_final.head())

(10158, 2)
   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_CID
0                     0.0        525.0
1                     0.0       1329.0
2                     0.0       1676.0
3                     4.0       1688.0
4                     0.0       1780.0


Again, this label set must be converted to a list prior to use in order to remove the index values 

In [28]:
lbl2_final_export = lbl2_final['PUBCHEM_ACTIVITY_SCORE']

lbl2_final_export.to_csv('labels_set_2.csv')

## Label Set 3:

Same sequence of steps will be followed 

In [29]:
lbl3 = df1
lbl3.head()

PUBCHEM_RESULT_TAG  PUBCHEM_SID  PUBCHEM_CID PUBCHEM_ACTIVITY_OUTCOME  \
0              29176    4243530.0        525.0                 Inactive   
1              10808     855876.0       1329.0                 Inactive   
2              10717     855784.0       1676.0                 Inactive   
3              10549     855615.0       1688.0                 Inactive   
4              10561     855627.0       1780.0                 Inactive   

   PUBCHEM_ACTIVITY_SCORE  PUBCHEM_ACTIVITY_URL  PUBCHEM_ASSAYDATA_COMMENT  \
0                     0.0                   NaN                        NaN   
1                     0.0                   NaN                        NaN   
2                     0.0                   NaN                        NaN   
3                     4.0                   NaN                        NaN   
4                     0.0                   NaN                        NaN   

  Inhibition_Primary at 6 uM  
0                       -3.2  
1                        0.9  
2                       -0.9  
3                        7.2  
4                       -2.3

In [30]:
IP6M = "Inhibition_Primary at 6 uM"  #set column name to variable for convenience
subset_lbl3 = [IP6M ,"PUBCHEM_CID"]
lbl3 = lbl3[subset_lbl3] #copy the kept columns to a new df called "lbl1_2" - Lbl2 is for Label set 2
print(lbl3.head()) #examine the df to confirm desired transformation took place"
print(lbl3.shape)

  Inhibition_Primary at 6 uM  PUBCHEM_CID
0                       -3.2        525.0
1                        0.9       1329.0
2                       -0.9       1676.0
3                        7.2       1688.0
4                       -2.3       1780.0
(10000, 2)


In [31]:
IP6M_active_dict = {}  

#Exact same process as above

#table_end is defined above (length of column + 10000)


for i in range(10000, table_end):
    if df2["PUBCHEM_ACTIVITY_OUTCOME"][i] == "Active":
        CID = df2["PUBCHEM_CID"][i]
        IP6M_score = df2[IP6M][i]
        IP6M_active_dict[CID] = IP6M_score
        
print(str(len(IP6M_active_dict)))  

158


In [39]:
comp_ids_list =  []
IP6M_list = []

for key, value in IP6M_active_dict.items():
    comp_ids_list.append(key)
    IP6M_list.append(value)

IP6M_set_data = {'PUBCHEM_CID': comp_ids_list, IP6M: IP6M_list}
active_df_IP6M = pd.DataFrame(IP6M_set_data) #df of active compounds, df contains CIDs and pscores


print(active_df_IP6M.head())
print(active_df_IP6M.shape)

  Inhibition_Primary at 6 uM  PUBCHEM_CID
0                       32.6    2533120.0
1                      128.6    3238146.0
2                       57.8    3237123.0
3                       27.2    5739478.0
4                       85.1    3236615.0
(158, 2)


#### Now to combine the two datasets (lbl3 + active_df_IP6M)  

In [41]:
lbl3.shape
lbl3.head(10)
lbl3_final = lbl3.append(active_df_IP6M)
print(lbl3_final.shape)
print(lbl3_final.head())

(10158, 2)
  Inhibition_Primary at 6 uM  PUBCHEM_CID
0                       -3.2        525.0
1                        0.9       1329.0
2                       -0.9       1676.0
3                        7.2       1688.0
4                       -2.3       1780.0


Convert df to list prior to use 

In [42]:
lbl3_final_export = lbl3_final[IP6M]

lbl3_final_export.to_csv('labels_set_3.csv')